web/client/components/share/SharePanel.jsx

```diff
@@ -241,7 +242,9 @@ class SharePanel extends React.Component {
         if (this.props.shareUrlRegex && this.props.shareUrlReplaceString) {
             shareEmbeddedUrl = this.generateUrl(shareEmbeddedUrl, this.props.shareUrlRegex, this.props.shareUrlReplaceString);
         }
-        const currentTab = !this.props.embedPanel && this.state.eventKey === 3 ? 1 : this.state.eventKey; // fallback to tab link if embed is disabled and selected at the same time
+        const {component: Permalink} = this.props.items?.find(({name} = {}) => name === "Permalink") ?? {};
+        const permalink = Permalink ? <Permalink shareUrl={shareUrl}/> : null;
+        const currentTab = ((!this.props.embedPanel && this.state.eventKey === 4) || (!permalink && this.state.eventKey === 3)) ? 1 : this.state.eventKey; // fallback to tab link if embed|permalink is disabled and selected at the same time
```

Comment:
Here I'd use `items` with `target` `tabs` to inject the item, without any specific reference to permalinks. 

In general Share plugin should not know anything about the injected tabs
. So the Item should pass the properties (e.g. title) and component, and the eventKey should be incremental (or passed). 
If embed is the last panel `eventKey` of embedPanel should be `2 + items.filter(({target})=>target === 'tabs').length` (or simply something like 10000).